File: pkg/definitions/updaters/javascript/fixtures/all.out.airplane.js
import airplane from "airplane";

export default airplane.task(
  {
    slug: "my_task_2",
    name: "Task name",
    description: "Task description",
    parameters: {
      dry: {
        name: "Dry run?",
        description: "Whether or not to run in dry-run mode.",
        type: "boolean",
        required: false,
        default: true
      },
      datetime: {
        type: "datetime",
        required: false
      }
    },
    runtime: "workflow",
    resources: ["db"],
    envVars: {
      CONFIG: {
        config: "aws_access_key"
      },
      VALUE: {
        value: "Hello World!"
      }
    },
    timeout: 60,
    constraints: {
      cluster: "k8s",
      vpc: "tasks"
    },
    requireRequests: true,
    allowSelfApprovals: false,
    restrictCallers: ["view", "task"],
    concurrencyKey: "scripts",
    concurrencyLimit: 5,
    schedules: {
      all: {
        name: "All fields",
        description: "A description",
        cron: "0 12 * * *",
        paramValues: {
          datetime: new Date("2006-01-02T15:04:05Z07:00")
        }
      },
      min: {
        cron: "* * * * *"
      }
    },
    permissions: {
      viewers: {
        groups: ["group1"],
        users: ["user1"]
      },
      requesters: {
        groups: ["group2"]
      },
      executers: {
        groups: ["group3", "group4"]
      },
      admins: {
        groups: ["group5"]
      }
    },
    defaultRunPermissions: "task-participants"
  },
  async () => {
    return []
  },
);

File: pkg/definitions/updaters/python/fixtures/all_out_airplane.py
from datetime import datetime, timezone
from typing import Optional
from typing_extensions import Annotated
import airplane


@airplane.task(
    name="Task name",
    description="Task description",
    resources=[airplane.Resource("db")],
    env_vars=[
        airplane.EnvVar(name="CONFIG", config_var_name="aws_access_key"),
        airplane.EnvVar(name="VALUE", value="Hello World!"),
    ],
    timeout=60,
    constraints={"cluster": "k8s", "vpc": "tasks"},
    require_requests=True,
    allow_self_approvals=False,
    restrict_callers=["view", "task"],
    schedules=[
        airplane.Schedule(
            slug="all",
            name="All fields",
            description="A description",
            cron="0 12 * * *",
            param_values={
                "datetime": datetime(2006, 1, 2, 15, 4, 5, tzinfo=timezone.utc)
            },
        ),
        airplane.Schedule(slug="min", cron="* * * * *"),
    ],
)
def my_task_2(
    dry: Annotated[
        Optional[bool],
        airplane.ParamConfig(
            name="Dry run?", description="Whether or not to run in dry-run mode."
        ),
    ] = True,
    datetime: Optional[datetime] = None,
):
    pass
